osquery/tables/system/windows/windows_security_products.cpp

```diff
@@ -0,0 +1,220 @@
+#include <codecvt>
+#include <locale>
+
+#include <iwscapi.h>
+#include <wscapi.h>
+
+#include <osquery/core.h>
+#include <osquery/core/windows/wmi.h>
+#include <osquery/logger.h>
+#include <osquery/tables.h>
+
+typedef struct wsc_entry {
+  WSC_SECURITY_PROVIDER provider;
+  std::wstring product_name;
+  WSC_SECURITY_PRODUCT_STATE product_state;
+  std::wstring product_state_timestamp;
+  std::wstring remediation_path;
+  WSC_SECURITY_SIGNATURE_STATUS signature_status;
+} wsc_entry;
+
+HRESULT GetSecurityProducts(WSC_SECURITY_PROVIDER provider,
+                            std::list<wsc_entry>& out_list) {
+  // Much of the following is adapted from the MS example at
+  // https://github.com/Microsoft/Windows-classic-samples/blob/master/Samples/WebSecurityCenter/cpp/WscApiSample.cpp
+
+  HRESULT hr = S_OK;
+  IWscProduct* PtrProduct = nullptr;
+  IWSCProductList* PtrProductList = nullptr;
+  BSTR PtrVal = nullptr;
+  LONG ProductCount = 0;
+  WSC_SECURITY_PRODUCT_STATE ProductState;
+  WSC_SECURITY_SIGNATURE_STATUS SignatureStatus;
+
+  if (provider != WSC_SECURITY_PROVIDER_FIREWALL &&
+      provider != WSC_SECURITY_PROVIDER_ANTIVIRUS &&
+      provider != WSC_SECURITY_PROVIDER_ANTISPYWARE) {
+    VLOG(1) << "Invalid security provider code: " << provider;
+    hr = E_INVALIDARG;
+    goto exit;
```

Comment:
May I ask you to use `scope_guard::Guard` from `osquery/utils/scope_guard.h` Instead of this C-way to manage resources. Or `std::unique_ptr` - if you find it better of easies for this case.